lnrpc/lightning.proto

```diff
@@ -2638,6 +2638,12 @@ message QueryRoutesRequest {
     fallback.
     */
     repeated lnrpc.FeatureBit dest_features = 17;
+
+    /*
+    The time preference for this payment. Set to -1 to optimize for fees
+    only, to 1 to optimize for reliability only or a value inbetween for a mix.
```

Comment:
Chose range [-1, 1] so that the default of 0 results in the current behavior. Perhaps [0, 1] would have been more logical for time preference? You have such preference (1) or you don't (0). 

On the RPC, it would require an extra bool `time_pref_specified` to not let the default be 0.

Reply:
Yeah the removal of the existing default values in the proto specification makes it harder at times to select a proper default value. 

I like how this is available as a per payment option, but what about also making this a global mission control parameter as well? With this route things continue to be uniform (the mission control API is where you specify any changes to the algo knobs), and it also lets application/servers just set the value once in the config (or at runtime) rather than always setting it each time a payment needs to go out (though the runtime value here can stay to allow per-payment tuning, but maybe now requires that bool depending on the default value?).

Reply:
I see where you're coming from, but in the end a global `time_pref` parameter is redundant. I think that by multiplying both `attemptcost` and `attemptcostppm` by that new term `(1/(0.5-timePref/2) - 1)`, the same effect can be achieved without adding the actual global `time_pref` parameter.

This would move the neutral point of `time_pref` on the payment level though. But I guess it isn't stable anyway because that point is ultimately defined by `attemptcost` and `attemptcostppm` regardless of whether there is also a global `time_pref`.

More generally, I don't think that global defaults are always a good thing. What we did with the fee limit for example in the payment v2 api, I think that turned out really well. Forcing users to decide on a fee limit rather than silently going with an arbitrary percentage. For `time_pref` the same thing may apply.

The path forward that I see is to eventually remove `attemptcost` and `attemptcostppm` completely. Either choose hard defaults that are not exposed or use a simpler formula internally to factor in reliability. What is left is just an abstract `time_pref` payment-level option as the single control lever. If we'd later on replace the entire probability estimation engine with something different, `time_pref` can probably still remain that single control lever.

Reply:
No not a new parameter. I mean that `attemptcost` is already a global parameter which users can tune. Instead of adding a `timepref` parameter that would for example multiply `attemptcost` by 2, they could just as well double the `attemptcost` parameter itself.